libosmscout-map/include/osmscoutmap/LabelLayouter.h

```diff
@@ -786,18 +788,54 @@ constexpr bool debugLabelLayouter = false;
       std::vector<Glyph<NativeGlyph>> glyphs = label->ToGlyphs();
 
       double pathLength=labelPath.GetLength();
-      size_t countLabels=(pathLength-labelData.contourLabelSpace)/(label->width+labelData.contourLabelSpace);
+      double minimalSpace=2*labelData.contourLabelOffset;
+      size_t workingLabelCount  =0;
```

Comment:
How many iterations do we need for long ways, like ferry lines? It is good idea to start just with one label?

Reply:
Good question, since I haven't tried yet. The *increment* increases with *2, so the iteration get to huge values fast! Do you have an example way that worries you?

Reply:
you are right, it should be fast and by manual testing, I do not see any impact.